nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/TwitterStreamAPI.java

```diff
@@ -0,0 +1,174 @@
+    private InputStream stream;
+
+    private Set<String> parseCommaSeparatedListPropreties(final ProcessContext context, final String propertyName) {
+        Set<String> fields = null;
+        if (context.getProperty(propertyName).isSet()) {
+            fields = new HashSet<>();
+            final String fieldsString = context.getProperty(propertyName).getValue();
+            for (final String field: fieldsString.split(",")) {
+                fields.add(field.trim());
+            }
+        }
+        return fields;
+    }
+
+    public TwitterStreamAPI(final ProcessContext context, final BlockingQueue<String> queue, final ComponentLog logger) {
+        assert context != null;
+        assert queue != null;
+        assert logger != null;
+
+        this.queue = queue;
+        this.logger = logger;
+
+        this.tweetFields = parseCommaSeparatedListPropreties(context, TWEET_FIELDS_PROPERTY_NAME);
+        this.userFields = parseCommaSeparatedListPropreties(context, USER_FIELDS_PROPERTY_NAME);
+        this.mediaFields = parseCommaSeparatedListPropreties(context, MEDIA_FIELDS_PROPERTY_NAME);
+        this.pollFields = parseCommaSeparatedListPropreties(context, POLL_FIELDS_PROPERTY_NAME);
+        this.placeFields = parseCommaSeparatedListPropreties(context, PLACE_FIELDS_PROPERTY_NAME);
+        this.expansions = parseCommaSeparatedListPropreties(context, EXPANSIONS_PROPERTY_NAME);
+        this.backfillMinutes = context.getProperty(BACKFILL_MINUTES_PROPERTY_NAME).asInteger();
+
+        TwitterCredentialsBearer creds = new TwitterCredentialsBearer(context.getProperty(BEARER_TOKEN_PROPERTY_NAME).getValue());
+        api = new TwitterApi();
+        api.setTwitterCredentials(creds);
+
+        this.executorService = Executors.newSingleThreadExecutor();
+    }
+
+    public String getBasePath() {
+        return api.getApiClient().getBasePath();
+    }
+
+    public void setBasePath(final String path) {
+        api.getApiClient().setBasePath(path);
+    }
+
+    /**
+     * This method would be called when we would like the stream to get started. This method will spin off a thread that
+     * will continue to queue tweets on to the given queue passed in the constructor. The thread will continue
+     * to run until {@code stop} is called.
+     * @param endpoint {@code TwitterStreamAPI.SAMPLE_ENDPOINT} or {@code TwitterStreamAPI.SEARCH_ENDPOINT}
+     */
+    public void start(final String endpoint) {
+        try {
+            if (endpoint.equals(SAMPLE_ENDPOINT)) {
+                stream = api.tweets().sampleStream(expansions, tweetFields, userFields, mediaFields, placeFields, pollFields, backfillMinutes);
+            } else if (endpoint.equals(SEARCH_ENDPOINT)) {
+                stream = api.tweets().searchStream(expansions, tweetFields, userFields, mediaFields, placeFields, pollFields, backfillMinutes);
+            } else {
+                throw new ProcessException("Endpoint was invalid value: " + endpoint);
+            }
```

Comment:
Instead of checking each value and throwing what should be an `IllegalStateException` in the `else` case, recommend having an conditional check for the `SAMPLE_ENDPOINT`, and defaulting the the search endpoint in the else case.